core/src/main/java/org/infinispan/configuration/cache/TransactionConfigurationBuilder.java

```diff
@@ -327,6 +332,18 @@ public void validate() {
       if (attributes.attribute(TRANSACTION_MODE).get() == TransactionMode.TRANSACTIONAL && !cacheMode.isSynchronous()) {
          throw log.unsupportedAsyncCacheMode(cacheMode);
       }
+      if (attributes.attribute(TRANSACTION_MODE).get() == TransactionMode.TRANSACTIONAL &&
+            attributes.attribute(LOCKING_MODE).get() == LockingMode.OPTIMISTIC &&
+            !attributes.attribute(USE_EAGER_LOCKING).get() && //this is pessimistic transaction
+            getBuilder().locking().isolationLevel() == IsolationLevel.REPEATABLE_READ &&
+            cacheMode != CacheMode.INVALIDATION_SYNC && //write-skew only for REPL, DIST and LOCAL (async is not supported in tx mode)
+            !getBuilder().invocationBatching().isEnabled()) { //batching can live without WSC
+         getBuilder().locking().writeSkewCheck(true);
+         getBuilder().versioning().enable().scheme(VersioningScheme.SIMPLE);
```

Comment:
I'm not sure about this, I think it will look as if it was modified by the user and it will mess with configuration inheritance. Maybe we can mark the new attribute value as "implicit" so it's not inherited? Otherwise I'd leave them disabled and do this check in a `Configurations.isWriteSkewEnabled()` method.